cardano-cli/src/Cardano/CLI/Run.hs

```diff
@@ -161,20 +163,22 @@ runDisplayVersion = do
  where
   renderVersion = Text.pack . showVersion
 
-helpAll :: ParserPrefs -> String -> [String] -> ParserInfo a -> IO ()
-helpAll pprefs progn rnames parserInfo = do
+helpAll :: OptVisibility -> ParserPrefs -> String -> [String] -> ParserInfo a -> IO ()
+helpAll visibility pprefs progn rnames parserInfo = do
   IO.putStrLn $ customRenderHelp 80 (usage_help parserInfo)
   IO.putStrLn ""
   go (infoParser parserInfo)
  where
   go :: Parser a -> IO ()
   go p = case p of
     NilP _ -> return ()
-    OptP optP -> case optMain optP of
-      CmdReader _ cs -> do
-        forM_ cs $ \(c, subParserInfo) ->
-          helpAll pprefs progn (c : rnames) subParserInfo
-      _ -> return ()
+    OptP optP ->
+      let visible = propVisibility (optProps optP) >= visibility
```

Comment:
From the point of view of a user facing parameter, having 3 values is what makes most sense, indeed.

But from the point of view of the golden files, an issue I see is that `help-internal.cli` includes `help-hidden.cli` which includes `help.cli`, which is a lot of repeated stuff.

And idea is we could have something like unix permissions, so that, for example:
- 1 = visible (x)
- 2 = hidden (w)
- 3 = visible + hidden (rw)
- 4 = internal (x)
...
- 7 = everything  (rwx)

A bit less intuitive though